libqtile/widget/__init__.py

```diff
@@ -37,66 +39,85 @@
+    'cmus': ['Cmus'],
+    'wallpaper': ['Wallpaper'],
+    'check_updates': ['CheckUpdates'],
+    'moc': ['Moc'],
+    'memory': ['Memory'],
+    'cpu': ['CPU'],
+    'idlerpg': ['IdleRPG'],
+    'pomodoro': ['Pomodoro'],
+    'stock_ticker': ['StockTicker'],
+    'caps_num_lock_indicator': ['CapsNumLockIndicator'],
+    'quick_exit': ['QuickExit'],
+    'pulse_volume': ['PulseVolume']
+}
+
+
+__all__ = list(set.union(*(set(x) for x in LAZY_IMPORT_SPEC.values())))
+
 
 def safe_import(module_name, class_name):
     safe_import_(
         (".widget", module_name), class_name, globals(), fallback=make_error
     )
 
 
-safe_import("backlight", "Backlight")
-safe_import("battery", ["Battery", "BatteryIcon"])
-safe_import("currentscreen", "CurrentScreen")
-safe_import("debuginfo", "DebugInfo")
-safe_import(
-    "graph",
-    [
-        "CPUGraph",
-        "MemoryGraph",
-        "SwapGraph",
-        "NetGraph",
-        "HDDGraph",
-        "HDDBusyGraph",
-    ],
-)
-safe_import("maildir", "Maildir")
-safe_import("notify", "Notify")
-safe_import("sensors", "ThermalSensor")
-safe_import("sep", "Sep")
-safe_import("she", "She")
-safe_import("spacer", "Spacer")
-safe_import("generic_poll_text", ["GenPollText", "GenPollUrl"])
-safe_import("volume", "Volume")
-safe_import("windowtabs", "WindowTabs")
-safe_import("keyboardlayout", "KeyboardLayout")
-safe_import("df", "DF")
-safe_import("image", "Image")
-safe_import("gmail_checker", "GmailChecker")
-safe_import("clipboard", "Clipboard")
-safe_import("countdown", "Countdown")
-safe_import("tasklist", "TaskList")
-safe_import("pacman", "Pacman")
-safe_import("launchbar", "LaunchBar")
-safe_import("canto", "Canto")
-safe_import("mpriswidget", "Mpris")
-safe_import("mpris2widget", "Mpris2")
-safe_import("mpd2widget", "Mpd2")
-safe_import("yahoo_weather", "YahooWeather")
-safe_import("bitcoin_ticker", "BitcoinTicker")
-safe_import("wlan", "Wlan")
-safe_import("khal_calendar", "KhalCalendar")
-safe_import("imapwidget", "ImapWidget")
-safe_import("net", "Net")
-safe_import("keyboardkbdd", "KeyboardKbdd")
-safe_import("cmus", "Cmus")
-safe_import("wallpaper", "Wallpaper")
-safe_import("check_updates", "CheckUpdates")
-safe_import("moc", "Moc")
-safe_import("memory", "Memory")
-safe_import("cpu", "CPU")
-safe_import("idlerpg", "IdleRPG")
-safe_import("pomodoro", "Pomodoro")
-safe_import("stock_ticker", "StockTicker")
-safe_import("caps_num_lock_indicator", "CapsNumLockIndicator")
-safe_import("quick_exit", "QuickExit")
-safe_import("pulse_volume", "PulseVolume")
+def __getattr__(name):
+    module_name, class_names = None, None
+    if name in __all__:
+        for mod, names in LAZY_IMPORT_SPEC.items():
+            if name in names:
+                module_name, class_names = mod, names
+                break
+    if (module_name, class_names) != (None, None):
+        safe_import(module_name, class_names)
+        if name in globals():
+            return globals()[name]
```

Comment:
Maybe these two lines could be moved to the top of the function to make look-ups of already imported objects faster...